### PR TITLE
Adds a list of Best Practices for Plugin Authors 

### PR DESCRIPTION
**Pull request by aditjind**
### Description
This PR adds a list of best practices that should be followed by the plugin developers.

 
### Issues Resolved
https://github.com/opensearch-project/opensearch-plugins/issues/65
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: BEST_PRACTICES.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Follow these best practices while developing any plugin:
```
Comment (dblock) [inline BEST_PRACTICES.md:1]: I think this should just be rolled into STANDARDS, cause those are all "best practices" (e.g. using a link checker). I would call this section "Development Best Practices".

**Review by dblock — COMMENTED** [review]
Thanks! Is this for OpenSearch Plugins, OpenSearch Dashboard Plugins or both? I think we want sections specifically for both. Also let's merge this into STANDARDS?

File: BEST_PRACTICES.md (hunk 1 above)
Comment (aditjind) [reply BEST_PRACTICES.md]: > Thanks! Is this for OpenSearch Plugins, OpenSearch Dashboard Plugins or both? I think we want sections specifically for both. Also let's merge this into STANDARDS?
> 
> For every section ideally we should have "good" vs. "bad", or links to PRs that show how something is done well.

This is true for any plugin development which takes place. Ideally the design principles mentioned here are for the backend app development but these best practices can be followed for dashboards as well.

I will circle back once I have PRs demonstrating these design principles and add them here.

File: STANDARDS.md [hunk 2]
```diff
@@ -3,13 +3,42 @@
...
   - [Code Coverage Report](#code-coverage-report)
   - [License Headers](#license-headers)
   - [Public Documentation](#public-documentation)
   - [Release Notes](#release-notes)  
 
+## Development Best Practices
+
+Follow these best practices while developing any plugin:
+
+#### Separation of Concerns
```
Comment (dblock) [inline STANDARDS.md:18]: These should be `###` and not `####` to nest under the topic above. And use a VSCode plugin to update the TOC?

File: STANDARDS.md [hunk 3]
```diff
@@ -3,13 +3,42 @@
...
   - [Broken Links Checker](#broken-links-checker)
   - [CI Workflows](#ci-workflows)
   - [Code Coverage Report](#code-coverage-report)
   - [License Headers](#license-headers)
   - [Public Documentation](#public-documentation)
   - [Release Notes](#release-notes)  
 
+## Development Best Practices
+
+Follow these best practices while developing any plugin:
```
Comment (dblock) [inline STANDARDS.md:16]: Nitpick, change `:` to `.`.

**Review by dblock — CHANGES_REQUESTED** [review]
I like this! Fix the TOC and un-draft it.

File: STANDARDS.md (hunk 3 above)
Comment (aditjind) [reply STANDARDS.md]: Done

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Merged, thanks.

**Pull request merged**